lib/filter.js

```diff
@@ -0,0 +1,86 @@
+'use strict';
+
+//////////////////////////////
+// contains, is, isNot, doesNotContain, exists, matches
+//
+// Include
+//////////////////////////////
+const addModule = function (results, pkg) {
+  try {
+    results[pkg.package.name] = require(pkg.package.name);
+  }
+  catch (e) {
+    if (!results.hasOwnProperty('__unrequireable')) {
+      results.__unrequireable = [];
+    }
+    results.__unrequireable.push(pkg.package.name);
+  }
+  return results;
+}
+
+module.exports = function ModuleFilter(packages, key) {
+  /**
+    * Contains
+  **/
+  this.containsSync = (search) => {
+    var results = {};
```

Comment:
Also, you're not going to win arguments by saying "best practices say use this" when both ways are equally valid for the required output. If we were exposing an array at the end, `.map` would be fine, but we are not.